### ISSUE TITLE
Clean up old imports in standard library

### ISSUE THREAD

**Issue by mlutze** [flix/flix#8096]
```bash
grep -rI "import" | cut -f1 -d: | grep "flix$" | sort -u | awk '{ print ("- [ ] " $0) }'
```

- [x] examples/using-laziness-for-logging.flix
- [x] main/src/library/Adaptor.flix
- [x] main/src/library/Add.flix
- [x] main/src/library/Benchmark.flix
- [x] main/src/library/BigDecimal.flix
- [x] main/src/library/BigInt.flix
- [x] main/src/library/Chain.flix
- [x] main/src/library/Char.flix
- [x] main/src/library/CodePoint.flix
- [x] main/src/library/Concurrent/Channel.flix
- [x] main/src/library/Concurrent/Condition.flix
- [x] main/src/library/Concurrent/CyclicBarrier.flix
- [x] main/src/library/Concurrent/ReentrantLock.flix
- [x] main/src/library/Console.flix
- [x] main/src/library/Debug.flix
- [x] main/src/library/Div.flix
- [x] main/src/library/Environment.flix
- [x] main/src/library/Eq.flix
- [x] main/src/library/File.flix
- [x] main/src/library/Files.flix
- [x] main/src/library/Fixpoint/Solver.flix
- [x] main/src/library/Float32.flix
- [x] main/src/library/Float64.flix
- [x] main/src/library/FromJava.flix
- [x] main/src/library/Hash.flix
- [x] main/src/library/Int16.flix
- [x] main/src/library/Int32.flix
- [x] main/src/library/Int64.flix
- [x] main/src/library/Int8.flix
- [x] main/src/library/IOError.flix
- [x] main/src/library/List.flix
- [x] main/src/library/Map.flix
- [x] main/src/library/Mul.flix
- [x] main/src/library/Nec.flix
- [x] main/src/library/Neg.flix
- [x] main/src/library/Nel.flix
- [x] main/src/library/Object.flix
- [x] main/src/library/Order.flix
- [x] main/src/library/PartialOrder.flix
- [x] main/src/library/Prelude.flix
- [x] main/src/library/Random.flix
- [x] main/src/library/Regex.flix
- [x] main/src/library/Result.flix
- [x] main/src/library/Set.flix
- [x] main/src/library/StringBuilder.flix
- [x] main/src/library/String.flix
- [x] main/src/library/Sub.flix
- [x] main/src/library/System.flix
- [x] main/src/library/Thread.flix
- [x] main/src/library/Time/Epoch.flix
- [x] main/src/library/Time/Instant.flix
- [x] main/src/library/ToJava.flix
- [x] main/src/library/ToString.flix
- [x] main/src/library/Vector.flix
- [x] main/test/ca/uwaterloo/flix/library/TestAdaptor.flix
- [x] main/test/ca/uwaterloo/flix/library/TestChar.flix
- [x] main/test/ca/uwaterloo/flix/library/TestFile.flix
- [x] main/test/ca/uwaterloo/flix/library/TestFloat32.flix
- [x] main/test/ca/uwaterloo/flix/library/TestFloat64.flix
- [x] main/test/ca/uwaterloo/flix/library/TestFromJava.flix
- [x] main/test/ca/uwaterloo/flix/library/TestInt16.flix
- [x] main/test/ca/uwaterloo/flix/library/TestInt32.flix
- [x] main/test/ca/uwaterloo/flix/library/TestInt64.flix
- [x] main/test/ca/uwaterloo/flix/library/TestInt8.flix
- [x] main/test/ca/uwaterloo/flix/library/TestToJava.flix
- [x] main/test/flix/Test.Exp.CheckedTypeCast.flix
- [x] main/test/flix/Test.Exp.Instanceof.flix
- [x] main/test/flix/Test.Exp.Interpolation.flix
- [x] main/test/flix/Test.Exp.Jvm.GetFieldDoubleNestedClass.flix
- [x] main/test/flix/Test.Exp.Jvm.GetField.flix
- [x] main/test/flix/Test.Exp.Jvm.GetFieldStaticInnerClass.flix
- [x] main/test/flix/Test.Exp.Jvm.GetStaticField.flix
- [x] main/test/flix/Test.Exp.Jvm.GetStaticFieldStaticInnerClass.flix
- [x] main/test/flix/Test.Exp.Jvm.InvokeConstructor2.flix
- [x] main/test/flix/Test.Exp.Jvm.InvokeConstructor.flix
- [x] main/test/flix/Test.Exp.Jvm.InvokeMethod2.Overload.Null.flix
- [x] main/test/flix/Test.Exp.Jvm.InvokeMethod2.Overload.Subtypes.flix
- [x] main/test/flix/Test.Exp.Jvm.InvokeMethod.flix
- [x] main/test/flix/Test.Exp.Jvm.InvokeStaticMethod2.flix
- [x] main/test/flix/Test.Exp.Jvm.InvokeStaticMethod.flix
- [x] main/test/flix/Test.Exp.Jvm.NewObject.flix
- [x] main/test/flix/Test.Exp.Jvm.PutField.flix
- [x] main/test/flix/Test.Exp.Jvm.PutStaticField.flix
- [x] main/test/flix/Test.Exp.Jvm.TryCatch.flix
- [x] main/test/flix/Test.Exp.UncheckedTypeCast.flix
- [x] main/test/flix/Test.Handler.Ask.AtomicInteger.flix
- [x] main/test/flix/Test.Handler.Gen.AtomicInteger.flix
- [x] main/test/flix/Test.Handler.MultiShot.AtomicInteger.flix
- [x] main/test/flix/Test.Handler.NewObject.flix
- [x] main/test/flix/Test.Handler.OneShot.AtomicInteger.flix
- [x] main/test/flix/Test.Handler.ZeroShot.AtomicInteger.flix
- [x] main/test/flix/Test.Import.flix
- [x] main/test/flix/Test.Java.Function.flix

**Comment by mlutze** [comment]
I'll claim Adaptor - Char to start

**Comment by magnus-madsen** [comment]
1. Try to keep the code changes small and idomatic.
2. Two style will be normal:

```
    pub def charAt(i: Int32, s: String): Char = unsafe s.charAt(i)
```

or

```
pub def charAt(i: Int32, s: String): Char = unsafe {
  ...
}
```

If there is more calls.

**Comment by chanattan** [comment]
I'm claiming all Int classes.

**Comment by magnus-madsen** [comment]
@mlutze Why did we omit static imports-- they should also be rewritten :)

**Comment by magnus-madsen** [comment]
@chanattan It would make sense for you to do the Float32 and Float64 modules also.

**Comment by mlutze** [comment]
> @mlutze Why did we omit static imports-- they should also be rewritten :)

updated

**Comment by JonathanStarup** [comment]
Ill do concurrent/

**Comment by dghosef** [comment]
I'll take the unclaimed in prelude -> system

**Comment by JonathanStarup** [comment]
ill do *Big* things

**Comment by JonathanStarup** [comment]
im doing files

**Comment by magnus-madsen** [comment]
Current status: Compilation failed with 486 error(s).

**Comment by magnus-madsen** [comment]
> im doing files

Wait for this PR to merge https://github.com/flix/flix/pull/8118

**Comment by mlutze** [comment]
I'll do Object and Time/*

**Comment by chanattan** [comment]
I'm taking all InvokeConstructor/InvokeMethod/InvokeStaticMethod files.

**Comment by magnus-madsen** [comment]
Current status: Compilation failed with 369 error(s).

**Comment by magnus-madsen** [comment]
@JonathanStarup @mlutze @chanattan @dghosef Alright everybody, lets try to get this done today :)

**Comment by JonathanStarup** [comment]
my worktime is pushed to the afternoon but ill get on it

**Comment by magnus-madsen** [comment]
> my worktime is pushed to the afternoon but ill get on it

By then I am sure only the easy modules will be left 😆

**Comment by mlutze** [comment]
I claim `Test.Exp.I*`

**Comment by mlutze** [comment]
I claim `*String*`

**Comment by mlutze** [comment]
I claim `*Handler*`

**Comment by dghosef** [comment]
I'll take everything alphabetically after and including `main/test/flix/Test.Exp.Jvm.NewObject.flix`

**Comment by mlutze** [comment]
> I'll take everything alphabetically after and including `main/test/flix/Test.Exp.Jvm.NewObject.flix`

NB: I'm doing all the ones with "Handler" in the name

**Comment by chanattan** [comment]
I claim
- `main/src/library/Chain.flix`
- `main/src/library/Char.flix`
- `main/src/library/CodePoint.flix`
- `main/src/library/Console.flix`
- `main/src/library/Debug.flix`

**Comment by chanattan** [comment]
> I'm taking all InvokeConstructor/InvokeMethod/InvokeStaticMethod files. Edit:
> 
>     * `Test.Exp.Jvm.InvokeConstructor2` doesn't need refactoring
> 
>     * `Test.Exp.Jvm.InvokeMethod2.Overload.Null` doesn't need refactoring

The work is still going on but no full support yet as there remains an array static call error. I am working on it in parallel.

**Comment by dghosef** [comment]
I'll take main/test/ca/uwaterloo/flix/library/Test*

**Comment by dghosef** [comment]
If anyone finds this helpful, https://docs.google.com/spreadsheets/d/19tAdmy6g9nfxJeI8R1oe16IYOIZB2ewXxhEtwLQ7qPk/edit?usp=sharing is a spreadsheet of the unclaimed files. I think it is correct/updated.

**Comment by dghosef** [comment]
I'll take file.flix

**Comment by magnus-madsen** [comment]
Current status: Compilation failed with 315 error(s).

(This is for the library itself)

**Comment by magnus-madsen** [comment]
@dghosef The get and put fields ones still have a lot of import static that can be refactored :)

**Comment by dghosef** [comment]
> @dghosef The get and put fields ones still have a lot of import static that can be refactored :)

If I change
    @test
    def testGetBoolField01(): Bool \ IO =
        import java_new dev.flix.test.TestClass(): ##dev.flix.test.TestClass \ IO as newObject;
        import java_get_field dev.flix.test.TestClass.boolField: Bool \ IO as getField;
        let o = newObject();
        getField(o) == true

into

    @test
    def testGetBoolField01(): Bool \ IO =
        import java_new dev.flix.test.TestClass(): ##dev.flix.test.TestClass \ IO as newObject;
        let o = newObject();
        o.getField() == true

I get a compilation error. I was under the impression that it can't be changed

**Comment by magnus-madsen** [comment]
We don't have field access yet, that is true.

**Comment by JonathanStarup** [comment]
ill do adapter

**Comment by magnus-madsen** [comment]
Current status: Compilation failed with 261 error(s).

**Comment by magnus-madsen** [comment]
I unchecked some modules because they still have refactorable imports.

**Comment by JonathanStarup** [comment]
im doing files, if not clear

**Comment by JonathanStarup** [comment]
> I unchecked some modules because they still have refactorable imports.

@magnus-madsen I checked some because they have waiting PRs, I though the list was useful to see whats not been started on

**Comment by mlutze** [comment]
I claim 
```
 main/test/ca/uwaterloo/flix/library/TestInt16.flix
main/test/ca/uwaterloo/flix/library/TestInt32.flix
main/test/ca/uwaterloo/flix/library/TestInt64.flix
main/test/ca/uwaterloo/flix/library/TestInt8.flix
```

**Comment by dghosef** [comment]
> I claim
> 
> ```
>  main/test/ca/uwaterloo/flix/library/TestInt16.flix
> main/test/ca/uwaterloo/flix/library/TestInt32.flix
> main/test/ca/uwaterloo/flix/library/TestInt64.flix
> main/test/ca/uwaterloo/flix/library/TestInt8.flix
> ```

I claimed those

**Comment by mlutze** [comment]
> > I claim
> > ```
> >  main/test/ca/uwaterloo/flix/library/TestInt16.flix
> > main/test/ca/uwaterloo/flix/library/TestInt32.flix
> > main/test/ca/uwaterloo/flix/library/TestInt64.flix
> > main/test/ca/uwaterloo/flix/library/TestInt8.flix
> > ```
> 
> I claimed those, but you are welcome to take them if you want ;)

I'll try to find something else

**Comment by mlutze** [comment]
Here are the current claims of unchecked tasks (I claimed the ones where I could not find another claim)

```
Jonathan        main/src/library/Adaptor.flix
Chanattan       main/src/library/CodePoint.flix
Chanattan       main/src/library/Console.flix
Joseph  main/src/library/Environment.flix
Jonathan        main/src/library/File.flix
Jonathan        main/src/library/Files.flix
Matt    main/src/library/Prelude.flix
Joseph  main/src/library/Regex.flix
Joseph  main/src/library/String.flix
Joseph  main/test/ca/uwaterloo/flix/library/TestFromJava.flix
Matt    main/test/flix/Test.Exp.CheckedTypeCast.flix
Joseph  main/test/flix/Test.Exp.Jvm.GetFieldDoubleNestedClass.flix
Joseph  main/test/flix/Test.Exp.Jvm.GetField.flix
Joseph  main/test/flix/Test.Exp.Jvm.GetFieldStaticInnerClass.flix
Joseph  main/test/flix/Test.Exp.Jvm.GetStaticField.flix
Joseph  main/test/flix/Test.Exp.Jvm.GetStaticFieldStaticInnerClass.flix
Chanattan       main/test/flix/Test.Exp.Jvm.InvokeConstructor.flix
Chanattan       main/test/flix/Test.Exp.Jvm.InvokeMethod2.Overload.Subtypes.flix
Chanattan       main/test/flix/Test.Exp.Jvm.InvokeMethod.flix
Chanattan       main/test/flix/Test.Exp.Jvm.InvokeStaticMethod2.flix
Chanattan       main/test/flix/Test.Exp.Jvm.InvokeStaticMethod.flix
Matt    main/test/flix/Test.Exp.Jvm.PutField.flix
Matt    main/test/flix/Test.Exp.Jvm.PutStaticField.flix
Matt    main/test/flix/Test.Exp.UncheckedTypeCast.flix
```

**Comment by JonathanStarup** [comment]
Is that all the remaining or is there any I can do now?

**Comment by chanattan** [comment]
Can we have an update on checkmarks?

**Comment by mlutze** [comment]
I've updated the checkmarks based on what I've seen tagged here.

**Comment by chanattan** [comment]
All the InvokeMethod or InvokeStaticMethod should be done.

**Comment by magnus-madsen** [comment]
Current status: Compilation failed with 55 error(s).

Missing: Array, CodePoint, Regex and String

**Comment by dghosef** [comment]
`main/test/flix/Test.Exp.Jvm.GetStaticField.flix` and
`main/test/flix/Test.Exp.Jvm.GetStaticFieldStaticInnerClass.flix` are finished

**Comment by magnus-madsen** [comment]
We have stuff left in `Array` and `Regex`

**Comment by chanattan** [comment]
I can work on Array.

**Comment by magnus-madsen** [comment]
> I can work on Array.

I think there is some advanced typematch stuff, so I will let Matt or Jonathan do it.

**Comment by magnus-madsen** [comment]
What tests are left?

**Comment by dghosef** [comment]
ill take regex

**Comment by dghosef** [comment]
Oh wait I remember - I intentionally didn't do the method calls in the regex that require a region effect since I thought we decided not to touch those.

**Comment by magnus-madsen** [comment]
> Oh wait I remember - I intentionally didn't do the method calls in the regex that require a region effect since I thought we decided not to touch those. Should I just cast them?

Yes, if you can make it work 👍

**Comment by chanattan** [comment]
InvokeConstructor should be already done also.

**Comment by magnus-madsen** [comment]
Yay! The Standard Library now compiles. Thanks everybody!!

As for the tests, there is still some work to do:

![image](https://github.com/user-attachments/assets/24967d9e-7c76-4acf-baf5-b161a91b9ac7)

@chanattan Could you have a look at these as your 1st priority? Thanks!

**Comment by mlutze** [comment]
Is this done?

**Comment by magnus-madsen** [comment]
No, some stuff remains.

**Comment by JonathanStarup** [comment]
@magnus-madsen what compilation flag are you using?

**Comment by magnus-madsen** [comment]
The deprecated flag

**Comment by magnus-madsen** [comment]
Fixed.
